apps/src/maze/bee.js

```diff
@@ -28,21 +28,33 @@ var Bee = function (maze, studioApp, config) {
   // honeycomb using an if block
   this.userChecks_ = [];
 
-  // The "raw dirt" represents the raw values as entered into the
-  // levelbuilder UI. They are a grid of either integers or strings,
-  // and can represent "variable" values, which means any given cell can
-  // theoretically contain one of a variety of objects.
+  // Initialize the map grid
   //
-  // We parse the strings into objects, then turn the "variable" grid
-  // into a list of "static" grids, representing all the possible
-  // combinations of objects.
+  // "serializedMaze" is the new way of storing maps; it's a JSON array
+  // containing complex map data.
   //
-  // We also initialize a "values" grid, which for objects that have
-  // values (hives and flowers) keeps track of their current state.
-  var rawDirt_ = config.level.rawDirt;
-  var variableGrid = rawDirt_.map(function (row) {
-    return row.map(Cell.parse);
-  });
+  // "map" plus optionally "levelDirt" is the old way of storing maps;
+  // they are each arrays of a combination of strings and ints with
+  // their own complex syntax. This way is deprecated for new levels,
+  // and only exists for backwards compatibility for not-yet-updated
+  // levels.
+  //
+  // Either way, we turn what we have into a grid of BeeCells, any one
+  // of which may represent a number of possible "static" cells. We then
+  // turn that variable grid of BeeCells into a set of static grids.
+  var variableGrid;
+  if (config.level.serializedMaze) {
+    variableGrid = config.level.serializedMaze.map(function (row) {
+      return row.map(BeeCell.deserialize);
+    });
+  } else {
+    variableGrid = config.level.map.map(function (row, x) {
+      return row.map(function (mapCell, y) {
+        var initialDirtCell = config.level.initialDirt[x][y];
+        return BeeCell.parseFromOldValues(mapCell, initialDirtCell);
+      });
+    });
+  }
```

Comment:
It looks like after the deserialization step here, we no longer need to think about the difference between old/new levels. That's great :) It also makes me think it *wouldn't* be super difficult to write a conversion script?